src/components/single-article.jsx

```diff
@@ -8,14 +8,21 @@ export const SingleArticle = () => {
   const [article, setArticle] = useState({});
   const [isLoading, setIsLoading] = useState(true);
   const articleId = useParams().article_id;
+  const [err, setErr] = useState(null);
 
   useEffect(() => {
-    fetchArticleById(articleId).then((data) => {
-      setArticle(data.article);
-      setIsLoading(false);
-    });
+    fetchArticleById(articleId)
+      .then((data) => {
+        setArticle(data.article);
+        setIsLoading(false);
+      })
+      .catch((err) => {
+        setErr("sorry could not load page, please try again");
```

Comment:
Might be nice to make your error messages a bit more specific to what the error is. for user feedback. If I were trying to go to an endpoint for something that didn't exist, and it told me to try again, I'd probably just keep refreshing the page.

Reply:
good point, i will make amends